lib/slop/result.rb

```diff
@@ -14,12 +14,22 @@ def initialize(parser)
       @options = parser.options
     end
 
-    # Returns an options value, nil if the option does not exist.
+    # Returns an option's value, nil if the option does not exist.
     def [](flag)
       (o = option(flag)) && o.value
     end
     alias get []
 
+    # Returns an option's value, raises KeyError if the option does not exist.
+    def fetch(flag)
+      o = option(flag)
```

Comment:
This could be simpler because we already support the functionality in `Result#[]`:

```ruby
def fetch(flag)
  get(flag) or raise KeyError, "key not found: '#{clean_key(flag)}'"
end
```

Also I don't think the message is clear enough. I would opt for something like "Could not find option with flag '#{flag}'". I don't think the flag needs cleaning, either, but I'm not too bothered either way. Also, any reason we shouldn't just re-use `Slop::UnknownOption` since it basically means the same thing?

Reply:
`get(flag) or raise KeyError, "key not found: '#{clean_key(flag)}'"` will throw an exception if the value of the key `flag` is nil this happens when you don't provide an option and that option does not have a default value as I test in _test/result_test.rb_:

```ruby
it "returns nil when an option is not provided and it does not have a default value" do
  @options.string("--hello")
  @result.parser.parse %w()

  assert_equal nil, @result.fetch('hello')
end
```

I tried to remove `clean_key(flag)` but if you try to make something like `@result.fetch("--unexisting")` then you'll receive the following error: `key not found: '--unexisting'`.

Reply:
> will throw an exception if the value of the key flag is nil this happens when you don't provide an option and that option does not have a default value as I test in test/result_test.rb

Right, good catch! 

> I tried to remove clean_key(flag) but if you try to make something like @result.fetch("--unexisting") then you'll receive the following error: key not found: '--unexisting'.

Yeah, I think that's okay. As I mentioned though, I don't feel too strongly about it so feel free to keep it. I'd be up for re-using `Slop::UnknownOption`, though

Reply:
> I'd be up for re-using Slop::UnknownOption, though

I think about it and you're right, so I make the changes for this!

I update the documentation for `Slop::UnknownOption` also so maybe you want to take a look at it:

> Raised when an unknown option is parsed or when trying to fetch an
unexisting option via `Slop::Result#fetch`.
Suppress with the `suppress_errors` config option.